modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/CategoriesListRowRedesignAdapter.kt

```diff
@@ -0,0 +1,127 @@
+package au.com.shiftyjelly.pocketcasts.discover.view
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat.getDrawable
+import androidx.core.content.ContextCompat.getString
+import androidx.recyclerview.widget.RecyclerView
+import au.com.shiftyjelly.pocketcasts.discover.R
+import au.com.shiftyjelly.pocketcasts.discover.databinding.CategoryPillBinding
+import au.com.shiftyjelly.pocketcasts.servers.model.DiscoverCategory
+import au.com.shiftyjelly.pocketcasts.servers.model.NetworkLoadableList
+
+@SuppressLint("NotifyDataSetChanged")
+class CategoriesListRowRedesignAdapter(
+    private val onCategoryClick: (NetworkLoadableList) -> Unit,
+    private val onAllCategoriesClick: (() -> Unit) -> Unit,
+    private val categories: List<CategoryPill>,
+) : RecyclerView.Adapter<CategoriesListRowRedesignAdapter.CategoriesRedesignViewHolder>() {
```

Comment:
Do you think we should consider to use ListAdapter rather than using RecyclerView.Adapter for better performance?

Reply:
I agree with you. The code is already updated. Thank you for your suggestion!